lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3051,6 +3167,29 @@ public static function remove_insecure_properties( $theme_json ) {
 					}
 
 					$variation_output = static::remove_insecure_styles( $variation_input );
+
+					// Process a variation's elements and element pseudo selector styles.
```

Comment:
The following block of code makes sure to allow elements within block style variations when removing insecure properties for a user without `unfiltered_html` capabilities e.g. an admin user on a multisite.

See `test_block_style_variations_with_elements_and_invalid_properties` in the unit tests.